packages/api-fetch/src/test/index.js

```diff
@@ -198,78 +201,61 @@ describe( 'apiFetch', () => {
 
 		controller.abort();
 
-		let error;
-
-		try {
-			await promise;
-		} catch ( err ) {
-			error = err;
-		}
-
-		expect( error.name ).toBe( 'AbortError' );
+		await expect( promise ).rejects.toMatchObject( {
+			name: 'AbortError',
+		} );
 	} );
 
-	it( 'should return null if response has no content status code', () => {
-		window.fetch.mockReturnValue(
-			Promise.resolve( {
-				status: 204,
-			} )
-		);
-
-		return apiFetch( { path: '/random' } ).catch( ( body ) => {
-			// eslint-disable-next-line jest/no-conditional-expect
-			expect( body ).toEqual( null );
+	it( 'should return null if response has no content status code', async () => {
+		globalThis.fetch.mockResolvedValue( {
+			ok: true,
+			status: 204,
 		} );
+
+		await expect( apiFetch( { path: '/random' } ) ).resolves.toBe( null );
```

Comment:
This test was wrong! It expected that `apiFetch` throws `null`, but in reality it returns `null`. The "204 No content" response doesn't have any body, so `apiFetch` returns `null`. And also avoids calling `response.json()` because that would throw.